src/core/space_interaction_API.jl

```diff
@@ -357,3 +359,48 @@ The value of the argument `r` and possible keywords operate identically to [`nea
 """
 nearby_agents(a, model, r = 1; kwargs...) =
     (model[id] for id in nearby_ids(a, model, r; kwargs...))
+
+"""
+    random_nearby_id(agent, model::ABM, r = 1; kwargs...) -> agent
+
+Return the `id` of a random agent near the position of the given `agent`.
+
+The value of the argument `r` and possible keywords operate identically to [`nearby_ids`](@ref).
+"""
+function random_nearby_id(a, model, r = 1; kwargs...)
+    # Uses Reservoir sampling (https://en.wikipedia.org/wiki/Reservoir_sampling)
+    iter = nearby_ids(a, model, r; kwargs...)
+
+    res = iterate(iter)
+    isnothing(res) && return    # `iterate` returns `nothing` when it ends
```

Comment:
We should also say that the function returns `nothing` if no agents are neargby.

Reply:
Ah, that would also make `random_nearby_agent` error. I'll fix that